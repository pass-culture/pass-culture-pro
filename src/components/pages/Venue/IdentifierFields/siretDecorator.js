import createDecorator from 'final-form-calculate'

import { getSireInfo, SIRET } from 'utils/sire'

const emptyValues = {
  address: null,
  city: null,
  latitude: null,
  longitude: null,
  name: null,
  postalCode: null,
}

export const createSiretDecorator = createDecorator({
  field: 'siret',
  updates: async siret => {
    const siretInfo = await getSireInfo(siret, SIRET)
    if (!siretInfo) return {}
    if (siretInfo.error) return emptyValues
    return siretInfo.values
  },
})

export default createSiretDecorator
