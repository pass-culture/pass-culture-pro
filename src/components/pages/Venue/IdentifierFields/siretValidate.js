import { getSireInfo, SIRET } from 'utils/siren'

export const siretValidate = async siret => {
  const siretInfo = await getSireInfo(siret, SIRET)
  if (!siretInfo || !siretInfo.error) return undefined
  return siretInfo.error
}

export default siretValidate
