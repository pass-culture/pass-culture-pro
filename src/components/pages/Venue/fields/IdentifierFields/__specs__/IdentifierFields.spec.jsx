import { shallow } from 'enzyme'
import React from 'react'
import IdentifierFields from '../IdentifierFields'
import TextareaField from '../../../../../layout/form/fields/TextareaField'
import TextField from '../../../../../layout/form/fields/TextField'

describe('src | components | pages | Venue | fields | IdentifierFields', () => {
  let props

  beforeEach(() => {
    props = {
      fieldReadOnlyBecauseFrozenFormSiret: true,
      formSiret: 'form siret',
      initialSiret: 'form siret',
      isCreatedEntity: true,
      isModifiedEntity: true,
      readOnly: true,
      venueTypes: [],
    }
  })

  it('should match the snapshot', () => {
    // when
    const wrapper = shallow(<IdentifierFields {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  describe('render', () => {
    it('should display four TextField and one TextAreaField components with the right props', () => {
      // given
      const props = {
        fieldReadOnlyBecauseFrozenFormSiret: false,
        initialSiret: null,
        isCreatedEntity: true,
        isModifiedEntity: true,
        readOnly: true,
        venueTypes: [],
      }

      // when
      const wrapper = shallow(<IdentifierFields {...props} />)

      // then
      const mainListTitle = wrapper.find('.main-list-title')
      expect(mainListTitle).toHaveLength(1)
      expect(mainListTitle.text()).toBe('Informations lieu')

      const textFields = wrapper.find(TextField)
      expect(textFields).toHaveLength(4)

      const textareaField = wrapper.find(TextareaField)
      expect(textareaField).toHaveLength(1)

      expect(textFields.at(0).prop('name')).toBe('siret')
      expect(textFields.at(0).prop('required')).toBe(false)

      expect(textFields.at(1).prop('label')).toBe('Nom du lieu : ')
      expect(textFields.at(1).prop('name')).toBe('name')
      expect(textFields.at(1).prop('required')).toBe(true)

      expect(textFields.at(2).prop('label')).toBe('Nom dâ€™usage du lieu : ')
      expect(textFields.at(2).prop('name')).toBe('publicName')
      expect(textFields.at(2).prop('required')).toBe(false)

      expect(textFields.at(3).prop('label')).toBe('E-mail : ')
      expect(textFields.at(3).prop('name')).toBe('bookingEmail')
      expect(textFields.at(3).prop('required')).toBe(true)

      expect(textareaField.at(0).prop('label')).toBe('Commentaire (si pas de SIRET) : ')
      expect(textareaField.at(0).prop('name')).toBe('comment')
    })

    describe('siret text field', () => {
      it('siret TextField can be edited when mode is not readOnly and there is no initial siret', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          initialSiret: null,
          readOnly: false,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(0).prop('name')).toBe('siret')
        expect(textFields.at(0).prop('readOnly')).toBe(false)
      })

      it('siret TextField cannot be edited when mode is read only', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          readOnly: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(0).prop('name')).toBe('siret')
        expect(textFields.at(0).prop('readOnly')).toBe(true)
      })

      it('proper siret label is returned when isCreatedEntity is true', () => {
        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const label = wrapper.find('label').at(0).text()
        expect(label).toBe('SIRET du lieu qui accueille vos offres (si applicable) : ')
      })

      it('proper siret label is returned when isCreatedEntity is false', () => {
        // given
        const props = {
          isCreatedEntity: false,
          isModifiedEntity: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const label = wrapper.find('label').at(0).text()
        expect(label).toBe('SIRET : ')
      })
    })

    describe('name text field', () => {
      it('name TextField can be edited when mode is not readOnly and fieldReadOnlyBecauseFrozenFormSiretdisplay is false', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          fieldReadOnlyBecauseFrozenFormSiret: false,
          readOnly: false,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(1).prop('name')).toBe('name')
        expect(textFields.at(1).prop('readOnly')).toBe(false)
      })

      it('name TextField cannot be edited when mode is read only', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          readOnly: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(1).prop('name')).toBe('name')
        expect(textFields.at(1).prop('readOnly')).toBe(true)
      })

      it('name TextField cannot be edited when fieldReadOnlyBecauseFrozenFormSiretdisplay is tue', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          fieldReadOnlyBecauseFrozenFormSiretdisplay: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(1).prop('name')).toBe('name')
        expect(textFields.at(1).prop('readOnly')).toBe(true)
      })
    })

    describe('publicName text field', () => {
      it('publicName TextField can be edited when mode is not readOnly', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          fieldReadOnlyBecauseFrozenFormSiret: false,
          readOnly: false,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(2).prop('name')).toBe('publicName')
        expect(textFields.at(2).prop('readOnly')).toBe(false)
      })

      it('publicName TextField cannot be edited when mode is read only', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          readOnly: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(2).prop('name')).toBe('publicName')
        expect(textFields.at(2).prop('readOnly')).toBe(true)
      })
    })

    describe('email text field', () => {
      it('email TextField can be edited when mode is not readOnly', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          fieldReadOnlyBecauseFrozenFormSiret: false,
          readOnly: false,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(3).prop('name')).toBe('bookingEmail')
        expect(textFields.at(3).prop('readOnly')).toBe(false)
      })

      it('email TextField cannot be edited when mode is read only', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          readOnly: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textFields = wrapper.find(TextField)
        expect(textFields.at(3).prop('name')).toBe('bookingEmail')
        expect(textFields.at(3).prop('readOnly')).toBe(true)
      })
    })

    describe('comment text area field', () => {
      it('comment text area field can be edited when mode is not readOnly', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          readOnly: false,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textareaField = wrapper.find(TextareaField)
        expect(textareaField.at(0).prop('name')).toBe('comment')
        expect(textareaField.at(0).prop('readOnly')).toBe(false)
      })

      it('comment text area field cannot be edited when mode is read only', () => {
        // given
        const props = {
          isCreatedEntity: true,
          isModifiedEntity: true,
          readOnly: true,
          venueTypes: [],
        }

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const textareaField = wrapper.find(TextareaField)
        expect(textareaField.at(0).prop('name')).toBe('comment')
        expect(textareaField.at(0).prop('readOnly')).toBe(true)
      })

      it('should not validate comment field when no siret have been provided', () => {
        // given
        props.formSiret = null

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const commentField = wrapper.find(TextareaField)
        expect(commentField.prop('validate')()).toBe('Ce champ est obligatoire')
      })

      it('should not validate empty comment field when siret length does not match 14 characters', () => {
        // given
        props.formSiret = 'AAA'

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const commentField = wrapper.find(TextareaField)
        expect(commentField.prop('validate')()).toBe('Ce champ est obligatoire')
      })

      it('should validate empty comment field when siret is provided and valid', () => {
        // given
        props.formSiret = '41816609600068'

        // when
        const wrapper = shallow(<IdentifierFields {...props} />)

        // then
        const commentField = wrapper.find(TextareaField)
        expect(commentField.prop('validate')()).toBe('')
      })
    })
  })
})
