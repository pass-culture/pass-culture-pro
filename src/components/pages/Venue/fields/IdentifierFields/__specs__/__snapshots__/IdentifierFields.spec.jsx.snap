// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Venue | fields | IdentifierFields should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IdentifierFields
    fieldReadOnlyBecauseFrozenFormSiret={true}
    formSiret="form siret"
    initialSiret="form siret"
    isCreatedEntity={true}
    isModifiedEntity={true}
    readOnly={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="main-list-title is-relative"
        >
          INFORMATIONS DU LIEU
        </h2>,
        <div
          className="field-group"
        >
          <HiddenField
            name="managingOffererId"
            validator={[Function]}
          />
          <div
            className="field text-field is-label-aligned"
          >
            <label
              className="field-label"
              htmlFor="siret"
            >
              <React.Fragment>
                SIRET
                <span
                  className="siret-label-details"
                >
                  <span
                    className="siret-label-bold"
                  >
                     du lieu qui accueille vos offres
                  </span>
                   (si applicable) : 
                </span>
              </React.Fragment>
            </label>
            <TextField
              className=""
              disabled={false}
              format={[Function]}
              id={null}
              innerClassName={null}
              isDecimal={true}
              label=""
              name="siret"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="siret"
              validate={null}
            />
          </div>
          <TextField
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName={null}
            isDecimal={true}
            label="Nom du lieu : "
            name="name"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={true}
            type="text"
            validate={null}
          />
          <TextField
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName={null}
            isDecimal={true}
            label="Nom d’usage du lieu : "
            name="publicName"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={false}
            type="text"
            validate={null}
          />
          <TextField
            className=""
            disabled={false}
            format={null}
            id={null}
            innerClassName="col-75"
            isDecimal={true}
            label="E-mail : "
            name="bookingEmail"
            parse={null}
            placeholder=""
            readOnly={true}
            renderInner={[Function]}
            renderValue={[Function]}
            required={true}
            type="email"
            validate={null}
          />
          <TextareaField
            autoComplete={false}
            className=""
            disabled={false}
            label="Commentaire (si pas de SIRET) : "
            maxLength={1000}
            name="comment"
            placeholder=""
            readOnly={true}
            required={false}
            rows={1}
            validate={[Function]}
            validating={false}
          />
        </div>,
      ],
      "className": "section",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "INFORMATIONS DU LIEU",
            false,
          ],
          "className": "main-list-title is-relative",
        },
        "ref": null,
        "rendered": Array [
          "INFORMATIONS DU LIEU",
          false,
        ],
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <HiddenField
              name="managingOffererId"
              validator={[Function]}
            />,
            <div
              className="field text-field is-label-aligned"
            >
              <label
                className="field-label"
                htmlFor="siret"
              >
                <React.Fragment>
                  SIRET
                  <span
                    className="siret-label-details"
                  >
                    <span
                      className="siret-label-bold"
                    >
                       du lieu qui accueille vos offres
                    </span>
                     (si applicable) : 
                  </span>
                </React.Fragment>
              </label>
              <TextField
                className=""
                disabled={false}
                format={[Function]}
                id={null}
                innerClassName={null}
                isDecimal={true}
                label=""
                name="siret"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={false}
                type="siret"
                validate={null}
              />
            </div>,
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              isDecimal={true}
              label="Nom du lieu : "
              name="name"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />,
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              isDecimal={true}
              label="Nom d’usage du lieu : "
              name="publicName"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="text"
              validate={null}
            />,
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-75"
              isDecimal={true}
              label="E-mail : "
              name="bookingEmail"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="email"
              validate={null}
            />,
            <TextareaField
              autoComplete={false}
              className=""
              disabled={false}
              label="Commentaire (si pas de SIRET) : "
              maxLength={1000}
              name="comment"
              placeholder=""
              readOnly={true}
              required={false}
              rows={1}
              validate={[Function]}
              validating={false}
            />,
          ],
          "className": "field-group",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "name": "managingOffererId",
              "validator": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="field-label"
                  htmlFor="siret"
                >
                  <React.Fragment>
                    SIRET
                    <span
                      className="siret-label-details"
                    >
                      <span
                        className="siret-label-bold"
                      >
                         du lieu qui accueille vos offres
                      </span>
                       (si applicable) : 
                    </span>
                  </React.Fragment>
                </label>,
                <TextField
                  className=""
                  disabled={false}
                  format={[Function]}
                  id={null}
                  innerClassName={null}
                  isDecimal={true}
                  label=""
                  name="siret"
                  parse={null}
                  placeholder=""
                  readOnly={true}
                  renderInner={[Function]}
                  renderValue={[Function]}
                  required={false}
                  type="siret"
                  validate={null}
                />,
              ],
              "className": "field text-field is-label-aligned",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <React.Fragment>
                    SIRET
                    <span
                      className="siret-label-details"
                    >
                      <span
                        className="siret-label-bold"
                      >
                         du lieu qui accueille vos offres
                      </span>
                       (si applicable) : 
                    </span>
                  </React.Fragment>,
                  "className": "field-label",
                  "htmlFor": "siret",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "SIRET",
                      <span
                        className="siret-label-details"
                      >
                        <span
                          className="siret-label-bold"
                        >
                           du lieu qui accueille vos offres
                        </span>
                         (si applicable) : 
                      </span>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "SIRET",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          <span
                            className="siret-label-bold"
                          >
                             du lieu qui accueille vos offres
                          </span>,
                          " (si applicable) : ",
                        ],
                        "className": "siret-label-details",
                      },
                      "ref": null,
                      "rendered": Array [
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "host",
                          "props": Object {
                            "children": " du lieu qui accueille vos offres",
                            "className": "siret-label-bold",
                          },
                          "ref": null,
                          "rendered": " du lieu qui accueille vos offres",
                          "type": "span",
                        },
                        " (si applicable) : ",
                      ],
                      "type": "span",
                    },
                  ],
                  "type": Symbol(react.fragment),
                },
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "",
                  "disabled": false,
                  "format": [Function],
                  "id": null,
                  "innerClassName": null,
                  "isDecimal": true,
                  "label": "",
                  "name": "siret",
                  "parse": null,
                  "placeholder": "",
                  "readOnly": true,
                  "renderInner": [Function],
                  "renderValue": [Function],
                  "required": false,
                  "type": "siret",
                  "validate": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": null,
              "isDecimal": true,
              "label": "Nom du lieu : ",
              "name": "name",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": true,
              "type": "text",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": null,
              "isDecimal": true,
              "label": "Nom d’usage du lieu : ",
              "name": "publicName",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": false,
              "type": "text",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "disabled": false,
              "format": null,
              "id": null,
              "innerClassName": "col-75",
              "isDecimal": true,
              "label": "E-mail : ",
              "name": "bookingEmail",
              "parse": null,
              "placeholder": "",
              "readOnly": true,
              "renderInner": [Function],
              "renderValue": [Function],
              "required": true,
              "type": "email",
              "validate": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "autoComplete": false,
              "className": "",
              "disabled": false,
              "label": "Commentaire (si pas de SIRET) : ",
              "maxLength": 1000,
              "name": "comment",
              "placeholder": "",
              "readOnly": true,
              "required": false,
              "rows": 1,
              "validate": [Function],
              "validating": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="main-list-title is-relative"
          >
            INFORMATIONS DU LIEU
          </h2>,
          <div
            className="field-group"
          >
            <HiddenField
              name="managingOffererId"
              validator={[Function]}
            />
            <div
              className="field text-field is-label-aligned"
            >
              <label
                className="field-label"
                htmlFor="siret"
              >
                <React.Fragment>
                  SIRET
                  <span
                    className="siret-label-details"
                  >
                    <span
                      className="siret-label-bold"
                    >
                       du lieu qui accueille vos offres
                    </span>
                     (si applicable) : 
                  </span>
                </React.Fragment>
              </label>
              <TextField
                className=""
                disabled={false}
                format={[Function]}
                id={null}
                innerClassName={null}
                isDecimal={true}
                label=""
                name="siret"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={false}
                type="siret"
                validate={null}
              />
            </div>
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              isDecimal={true}
              label="Nom du lieu : "
              name="name"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="text"
              validate={null}
            />
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName={null}
              isDecimal={true}
              label="Nom d’usage du lieu : "
              name="publicName"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={false}
              type="text"
              validate={null}
            />
            <TextField
              className=""
              disabled={false}
              format={null}
              id={null}
              innerClassName="col-75"
              isDecimal={true}
              label="E-mail : "
              name="bookingEmail"
              parse={null}
              placeholder=""
              readOnly={true}
              renderInner={[Function]}
              renderValue={[Function]}
              required={true}
              type="email"
              validate={null}
            />
            <TextareaField
              autoComplete={false}
              className=""
              disabled={false}
              label="Commentaire (si pas de SIRET) : "
              maxLength={1000}
              name="comment"
              placeholder=""
              readOnly={true}
              required={false}
              rows={1}
              validate={[Function]}
              validating={false}
            />
          </div>,
        ],
        "className": "section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "INFORMATIONS DU LIEU",
              false,
            ],
            "className": "main-list-title is-relative",
          },
          "ref": null,
          "rendered": Array [
            "INFORMATIONS DU LIEU",
            false,
          ],
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <HiddenField
                name="managingOffererId"
                validator={[Function]}
              />,
              <div
                className="field text-field is-label-aligned"
              >
                <label
                  className="field-label"
                  htmlFor="siret"
                >
                  <React.Fragment>
                    SIRET
                    <span
                      className="siret-label-details"
                    >
                      <span
                        className="siret-label-bold"
                      >
                         du lieu qui accueille vos offres
                      </span>
                       (si applicable) : 
                    </span>
                  </React.Fragment>
                </label>
                <TextField
                  className=""
                  disabled={false}
                  format={[Function]}
                  id={null}
                  innerClassName={null}
                  isDecimal={true}
                  label=""
                  name="siret"
                  parse={null}
                  placeholder=""
                  readOnly={true}
                  renderInner={[Function]}
                  renderValue={[Function]}
                  required={false}
                  type="siret"
                  validate={null}
                />
              </div>,
              <TextField
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName={null}
                isDecimal={true}
                label="Nom du lieu : "
                name="name"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={true}
                type="text"
                validate={null}
              />,
              <TextField
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName={null}
                isDecimal={true}
                label="Nom d’usage du lieu : "
                name="publicName"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={false}
                type="text"
                validate={null}
              />,
              <TextField
                className=""
                disabled={false}
                format={null}
                id={null}
                innerClassName="col-75"
                isDecimal={true}
                label="E-mail : "
                name="bookingEmail"
                parse={null}
                placeholder=""
                readOnly={true}
                renderInner={[Function]}
                renderValue={[Function]}
                required={true}
                type="email"
                validate={null}
              />,
              <TextareaField
                autoComplete={false}
                className=""
                disabled={false}
                label="Commentaire (si pas de SIRET) : "
                maxLength={1000}
                name="comment"
                placeholder=""
                readOnly={true}
                required={false}
                rows={1}
                validate={[Function]}
                validating={false}
              />,
            ],
            "className": "field-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "name": "managingOffererId",
                "validator": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <label
                    className="field-label"
                    htmlFor="siret"
                  >
                    <React.Fragment>
                      SIRET
                      <span
                        className="siret-label-details"
                      >
                        <span
                          className="siret-label-bold"
                        >
                           du lieu qui accueille vos offres
                        </span>
                         (si applicable) : 
                      </span>
                    </React.Fragment>
                  </label>,
                  <TextField
                    className=""
                    disabled={false}
                    format={[Function]}
                    id={null}
                    innerClassName={null}
                    isDecimal={true}
                    label=""
                    name="siret"
                    parse={null}
                    placeholder=""
                    readOnly={true}
                    renderInner={[Function]}
                    renderValue={[Function]}
                    required={false}
                    type="siret"
                    validate={null}
                  />,
                ],
                "className": "field text-field is-label-aligned",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <React.Fragment>
                      SIRET
                      <span
                        className="siret-label-details"
                      >
                        <span
                          className="siret-label-bold"
                        >
                           du lieu qui accueille vos offres
                        </span>
                         (si applicable) : 
                      </span>
                    </React.Fragment>,
                    "className": "field-label",
                    "htmlFor": "siret",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": Array [
                        "SIRET",
                        <span
                          className="siret-label-details"
                        >
                          <span
                            className="siret-label-bold"
                          >
                             du lieu qui accueille vos offres
                          </span>
                           (si applicable) : 
                        </span>,
                      ],
                    },
                    "ref": null,
                    "rendered": Array [
                      "SIRET",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": Array [
                            <span
                              className="siret-label-bold"
                            >
                               du lieu qui accueille vos offres
                            </span>,
                            " (si applicable) : ",
                          ],
                          "className": "siret-label-details",
                        },
                        "ref": null,
                        "rendered": Array [
                          Object {
                            "instance": null,
                            "key": undefined,
                            "nodeType": "host",
                            "props": Object {
                              "children": " du lieu qui accueille vos offres",
                              "className": "siret-label-bold",
                            },
                            "ref": null,
                            "rendered": " du lieu qui accueille vos offres",
                            "type": "span",
                          },
                          " (si applicable) : ",
                        ],
                        "type": "span",
                      },
                    ],
                    "type": Symbol(react.fragment),
                  },
                  "type": "label",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "",
                    "disabled": false,
                    "format": [Function],
                    "id": null,
                    "innerClassName": null,
                    "isDecimal": true,
                    "label": "",
                    "name": "siret",
                    "parse": null,
                    "placeholder": "",
                    "readOnly": true,
                    "renderInner": [Function],
                    "renderValue": [Function],
                    "required": false,
                    "type": "siret",
                    "validate": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": null,
                "isDecimal": true,
                "label": "Nom du lieu : ",
                "name": "name",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": true,
                "type": "text",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": null,
                "isDecimal": true,
                "label": "Nom d’usage du lieu : ",
                "name": "publicName",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": false,
                "type": "text",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "disabled": false,
                "format": null,
                "id": null,
                "innerClassName": "col-75",
                "isDecimal": true,
                "label": "E-mail : ",
                "name": "bookingEmail",
                "parse": null,
                "placeholder": "",
                "readOnly": true,
                "renderInner": [Function],
                "renderValue": [Function],
                "required": true,
                "type": "email",
                "validate": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "autoComplete": false,
                "className": "",
                "disabled": false,
                "label": "Commentaire (si pas de SIRET) : ",
                "maxLength": 1000,
                "name": "comment",
                "placeholder": "",
                "readOnly": true,
                "required": false,
                "rows": 1,
                "validate": [Function],
                "validating": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
